### PR TITLE
[Opensora-PKU] v1.2.0 training and inference

### PR DESCRIPTION
**Pull request by wtomin**
Support:
1. run vae reconstruction, t2i and t2v inference;
2. train stage 1 ~ 5;
3. Training with zero2, zero3, and sequence parallelism;
4. Training in pynative mode.
Overfitting experiment was conducted for stage 3 (29x480p). The overfitting loss curves and generated videos were similar to torch's overfitting experiment.

### CODE REVIEW

File: examples/opensora_pku/docs/structure.md
Comment (SamitHuang) [inline examples/opensora_pku/docs/structure.md:1]: update?

File: examples/opensora_pku/README.md [hunk 1]
```diff
@@ -417,152 +429,73 @@ python opensora/sample/sample_text_embed.py \
...
+python  opensora/train/train_t2v_diffusers.py \
+    --data "scripts/train_data/merge_data.txt" \
+    --num_frames ${NUM_FRAME} \
+    --max_height 480 \
+    --max_width 640 \
+    --attention_mode xformers \
+    --gradient_checkpointing \
+    --pretrained "path/to/ms-or-safetensors-ckpt/from/last/stage" \
+    --parallel_mode "zero" \
+    --zero_stage 2 \
```
Comment (CaitinZhao) [inline examples/opensora_pku/README.md:465]: multi cards?

File: examples/opensora_pku/opensora/train/train_t2v_diffusers.py [hunk 2]
```diff
@@ -364,56 +409,89 @@ def main(args):
...
+            latent_diffusion_with_loss.trainable_params(),
+            name=args.optim,
+            betas=args.betas,
+            eps=args.optim_eps,
+            group_strategy=args.group_strategy,
+            weight_decay=args.weight_decay,
+            lr=lr,
+        )
+    else:
+        from mindcv.optim.adamw import AdamW
```
Comment (SamitHuang) [inline examples/opensora_pku/opensora/train/train_t2v_diffusers.py:423]: just set `args.optim=adamw_re`

File: examples/opensora_pku/README.md [hunk 3]
```diff
@@ -2,12 +2,13 @@
 
 Here we provide an efficient MindSpore version of [Open-Sora-Plan](https://github.com/PKU-YuanGroup/Open-Sora-Plan/tree/main) from Peking University. We would like to express our gratitude to their contributions! :+1:
 
-**OpenSora-PKU is still under active development.** Currently, we are in line with **Open-Sora-Plan v1.1.0**.
+**OpenSora-PKU is still under active development.** Currently, we are in line with **Open-Sora-Plan v1.2.0**.
```
Comment (CaitinZhao) [inline examples/opensora_pku/README.md:5]: add a link to v1.1 branch v0.2.0

File: examples/opensora_pku/opensora/utils/ms_utils.py [hunk 4]
```diff
@@ -112,8 +133,8 @@ def init_env(
         ms.set_context(
             mode=mode,
             device_target=device_target,
-            ascend_config={"precision_mode": "allow_fp32_to_fp16"},
             device_id=device_id,
+            ascend_config={"precision_mode": "allow_fp32_to_fp16"},
```
Comment (SamitHuang) [inline examples/opensora_pku/opensora/utils/ms_utils.py:137]: why add? no mixed precision control in model construct

File: examples/opensora_pku/opensora/acceleration/communications.py [hunk 5]
```diff
@@ -62,7 +62,7 @@ def __init__(self, scatter_dim: int = 1, gather_dim: int = 0):
         super(AllToAll_SBH, self).__init__(scatter_dim=scatter_dim, gather_dim=gather_dim)
 
 
-def prepare_parallel_data(
+def prepare_parallel_data_bak(
```
Comment (CaitinZhao) [inline examples/opensora_pku/opensora/acceleration/communications.py:65]: why keep this?

File: examples/opensora_pku/opensora/dataset/loader.py [hunk 6]
```diff
@@ -1,91 +1,234 @@
+# Copyright 2023 Huawei Technologies Co., Ltd
```
Comment (CaitinZhao) [inline examples/opensora_pku/opensora/dataset/loader.py:1]: delete the copyright

File: examples/opensora_pku/opensora/dataset/loader.py [hunk 7]
```diff
@@ -1,91 +1,234 @@
...
 
 def create_dataloader(
-    ds_config,
+    dataset,
     batch_size,
-    ds_name="image",
+    column_names=["video"],
     num_parallel_workers=12,
-    max_rowsize=32,
+    max_rowsize=64,
```
Comment (CaitinZhao) [inline examples/opensora_pku/opensora/dataset/loader.py:34]: max_rowsize too large may lead to CPU momory OOM

File: examples/opensora_pku/opensora/dataset/transform.py [hunk 8]
```diff
@@ -1,4 +1,5 @@
 import html
+import random
```
Comment (CaitinZhao) [inline examples/opensora_pku/opensora/dataset/transform.py:2]: add random.seed at set_seed function

File: examples/opensora_pku/opensora/models/diffusion/opensora/net_with_loss.py [hunk 9]
```diff
@@ -204,43 +176,74 @@ def compute_loss(self, x, text_embed, encoder_attention_mask, attention_mask=Non
...
-        model_output, model_var_values = ops.split(model_output, C, axis=1)
-
-        # Learn the variance using the variational bound, but don't let it affect our mean prediction.
-        vb = self._cal_vb(ops.stop_gradient(model_output), model_var_values, x, x_t, t)
-
-        loss = mean_flat((noise - model_output) ** 2) + vb
-        loss = loss.mean()
+        bsz, c, _, _, _ = model_pred.shape
+        if attention_mask is not None:
+            attention_mask = attention_mask.unsqueeze(1).float().repeat(c, axis=1)  # b t h w -> b c t h w
```
Comment (CaitinZhao) [inline examples/opensora_pku/opensora/models/diffusion/opensora/net_with_loss.py:219]: why float?

File: examples/opensora_pku/README.md (hunk 3 above)
Comment (wtomin) [reply examples/opensora_pku/README.md:5]: I submitted a new PR to maintain opensora-pku v1.1.0 in branch v0.2.0. Waiting to be merged.

File: examples/opensora_pku/opensora/dataset/transform.py (hunk 8 above)
Comment (wtomin) [reply examples/opensora_pku/opensora/dataset/transform.py:2]: There is `random.seed` set in `set_random_seed`, imported from `mindone.utils.seed`

File: examples/opensora_pku/opensora/dataset/loader.py (hunk 7 above)
Comment (wtomin) [reply examples/opensora_pku/opensora/dataset/loader.py:34]: OK, changed to 32.

File: examples/opensora_pku/opensora/dataset/loader.py (hunk 6 above)
Comment (wtomin) [reply examples/opensora_pku/opensora/dataset/loader.py:1]: Deleted.

File: examples/opensora_pku/opensora/acceleration/communications.py (hunk 5 above)
Comment (wtomin) [reply examples/opensora_pku/opensora/acceleration/communications.py:65]: Deleted.

File: examples/opensora_pku/opensora/train/train_t2v_diffusers.py (hunk 2 above)
Comment (wtomin) [reply examples/opensora_pku/opensora/train/train_t2v_diffusers.py:423]: Fixed.

File: examples/opensora_pku/opensora/utils/ms_utils.py (hunk 4 above)
Comment (wtomin) [reply examples/opensora_pku/opensora/utils/ms_utils.py]: Removed

File: examples/opensora_pku/docs/structure.md
Comment (wtomin) [reply examples/opensora_pku/docs/structure.md:1]: Removed. Torch repo has removed this file too.

File: examples/opensora_pku/opensora/models/diffusion/opensora/net_with_loss.py (hunk 9 above)
Comment (wtomin) [reply examples/opensora_pku/opensora/models/diffusion/opensora/net_with_loss.py:219]: Torch code converts attention mask to float tensor here. I think it is to keep the multiplication of loss(fp32) and attention_mask (fp32) at the same data type: 
```python
loss = (loss * attention_mask).sum() / attention_mask.sum() 
```

File: examples/opensora_pku/README.md (hunk 3 above)
Comment (CaitinZhao) [reply examples/opensora_pku/README.md:5]: merged

**Review by SamitHuang — APPROVED** [review]

**Review by CaitinZhao — APPROVED** [review]

**Review by vigo999 — APPROVED** [review]
